tutorials/how_to/visualizing_model_internals.md

```diff
@@ -46,35 +50,54 @@ tokenize your data, and pass along the tokens there), but it was easiest this wa
-to show: `Collapsible` and `HeatMap`.  You just need to get the attention values you returned from
-`model.forward()` passed in to a `HeatMap` component with the right labels.  Again, with BiDAF,
-this was pretty straightforward.  Here we're modifying `McComponent.js`, because that's where the
-BiDAF rendering components are found (again see
-https://github.com/allenai/allennlp/pull/692/files):
+## Customize the react demo
 
-- We found the place where the request data was read and grabbed another field from the
-  returned json.  This looks something like `const attention = responseData &&
-responseData.passage_question_attention`, and similar for the `question_tokens` and
-`passage_tokens`.
-- We passed those fields in to the `McOutput` component, which renders the model output.
-- We modified the `McOutput` component to accept and render those fields.  We added two
-  `Collapsible` components, one wrapping all model internals, and one containing the attention
-heatmap.  The `HeatMap` component just takes the `question_tokens` as `xLabels`, the
-`passage_tokens` as `yLabels`, and the `attention` as `data`.  You can see the specific code in
-the linked pull request above.
+The demo code that you copied above has two files that you need to modify to make your demo work:
+`src/ModelInput.jsx` and `src/ModelOutput.jsx`.  `ModelInput.jsx` renders the left half of the
+screen, where you give inputs to the model, and `ModelOutput.jsx` renders the right half of the
+screen with the output of your model after you click on the "Run" button.
 
-That should be all you have to do to get model internals visualized in your demo.
+### Modifying ModelInput.jsx
 
-## Single-dimension heatmaps
+There are six `TODOs` in `ModelInput.jsx`, showing all of the places you might want to modify to
+get your demo hooked up correctly.  These modifications include changing the input elements to
+accept the inputs that your model expects, adding some canned examples that are easy to run in
+your demo, and giving a title and description for your model.  The most important point to make
+sure of is that the keys to the dictionary you pass to the `runModel` function match the keys that
+your model expects to receive in `predict_json`.
 
-The `HeatMap` component we used expects a two-dimensional data array.  If you have a
-one-dimensional data array you want to visualize, this is still possible, but you need to do
-something like `attention = attention.map(x => [x])` in the javascript to make your
-one-dimensional array into a two-dimensional array with length 1 on the x axis.  You also need to
-provide a single `xLabel` to the `HeatMap`, like `xLabels={['Probabilities']}`.
+Once you've modified these, you can check that you've done it correctly by clicking "Run" with an
+example and making sure that your simple server receives and can respond to the request.  You
+should be running the simple server on port 8000, and you should see the server log a `POST`
```

Comment:
I think this demo should use whatever the default port is for the simple_server, so either the demo should look for the backend on port 8888 (less work), or the simple_server should default to 8000 (which requires changing other tutorials, probably)

Reply:
Yeah, good point.  Any complaints about changing the default port to 8000?

Reply:
I did a grep through the repo (including tutorials) for `8888`, and there were only two lines to change, so I changed them.

Reply:
👍